ubidots/ubidots_out.js

```diff
@@ -1,49 +1,101 @@
-module.exports = function (RED) {
-  var mqtt = require('mqtt')
-
-  function postUbidots (self, endpointUrl, labelDevice, values, token) {
-    var client = mqtt.connect('mqtt://' + endpointUrl, { username: token, password: '' })
-
-    client.on('connect', function () {
-      client.publish(
-        '/v1.6/devices/' + labelDevice,
-        values,
-        { qos: 1, retain: false },
-        function () {
-          client.end(true)
-        }
-      )
-      self.status({ fill: 'green', shape: 'dot', text: 'ubidots.published' })
-    })
+module.exports = function(RED) {
+  var mqtt = require("mqtt");
+  var fs = require("fs");
+  var path = require("path");
 
-    client.on('error', function (msg) {
-      self.status({ fill: 'red', shape: 'ring', text: 'ubidots.error_connecting' })
-    })
-  }
+  function UbidotsNode(config) {
+    RED.nodes.createNode(this, config);
+    var self = this;
+    var ENDPOINT_URLS = {
+      business: "industrial.api.ubidots.com",
+      educational: "things.ubidots.com"
+    };
+    var useTLS = config.tls_checkbox;
+    var URL_PREFIX = "mqtt://";
 
-  function UbidotsNode (n) {
-    RED.nodes.createNode(this, n)
-    var self = this
+    var port = 1883;
+    var portTLS = 8883;
+    var certificate = fs.readFileSync(
+      path.join(__dirname, "./certificate.pem"),
```

Comment:
Why don't you use the certificate in the `keys` folder? This way we could prevent duplicating the file.